src/nvim/api/vim.c

```diff
@@ -46,6 +47,14 @@
 # include "api/vim.c.generated.h"
 #endif
 
+void nvim_source(String command, Error *err)
+  FUNC_API_SINCE(5)
+{
+    try_start();
+    do_source_str((char_u *)command.data);
+    update_screen(VALID);
```

Comment:
No need to invoke `update_screen`, it will be done after processing events (I will remove it from `nvim_command').